### DISCUSSION TITLE
The map file of dependency do not created with webpack5？

### DISCUSSION THREAD

**Discussion by BUPTlhuanyu** [webpack/webpack#15166] [Q&A] [answered]
#### Config：
```
devtool: 'source-map'
```
![image](https://user-images.githubusercontent.com/27213108/149487935-bf1c907a-53fe-48df-acc8-b9571d375218.png)

#### The dependency dist:
![image](https://user-images.githubusercontent.com/27213108/149480641-9f51beda-d4bc-4b89-9062-46292223b398.png)

#### the san.xxx.map.js not created
![image](https://user-images.githubusercontent.com/27213108/149482016-7330a21f-b1d1-4b42-97bd-212e0dd6a464.png)

**Comment by alexander-akait** [comment, marked as answer]
Please provide example of the problem

**Reply by BUPTlhuanyu** [reply]
It's hard to provide an example, But i found that when the `getTaskForFile` function (https://github.com/webpack/webpack/blob/b3871e84122b6cc5298ad403ca46c3cd3df2a2b5/lib/SourceMapDevToolPlugin.js#L272) of SourceMapDevToolPlugin handle the san.xxx.js ,  the options.finalSource 
https://github.com/webpack/webpack-sources/blob/9f98066311d53a153fdc7c633422a1d086528027/lib/ConcatSource.js#L101
 is always true, so that this callback 
https://github.com/webpack/webpack-sources/blob/9f98066311d53a153fdc7c633422a1d086528027/lib/helpers/getFromStreamChunks.js#L29
 wont execute.

**Reply by BUPTlhuanyu** [reply]
![image](https://user-images.githubusercontent.com/27213108/149513742-6b910c36-0d20-484c-9b16-23e01750ada8.png)
if I do that, the result will be the same as the webpack4 with webpack-source1.4.3

**Reply by BUPTlhuanyu** [reply]
haha! I found that i have created an empty map file! 
https://github.com/webpack/webpack-sources/commit/b802a5e71e39584b053bb573ae0360b0fa370d52

**Reply by BUPTlhuanyu** [reply]
It's not a problem at all!
